roofit/roofitcore/src/RooMinimizer.cxx

```diff
@@ -96,43 +97,81 @@ void RooMinimizer::cleanup()
 /// for HESSE and MINOS error analysis is taken from the defaultErrorLevel()
 /// value of the input function.
 
-RooMinimizer::RooMinimizer(RooAbsReal &function, FcnMode fcnMode) : _fcnMode(fcnMode)
+/// Construcor that accepts all configuration in struct with RooAbsReal likelihood
+RooMinimizer::RooMinimizer(RooAbsReal &function, Config &cfg) : _cfg(cfg)
 {
+   // Call init functions
    initMinimizerFirstPart();
+   if (_cfg.parallelGradient || _cfg.parallelLikelihood) {
+      RooFit::TestStatistics::RooRealL *nll_real = dynamic_cast<RooFit::TestStatistics::RooRealL *>(&function);
+      if (nll_real == nullptr)
+         throw std::logic_error("In RooMinimizer constructor: Selected parallel gradient or likelihood evaluation "
+                                "but an old-style likelihood was given. Please supply NewStyle(true) as an "
+                                "argument to createNLL for new-style likelihoods.");
+      std::shared_ptr<RooFit::TestStatistics::RooAbsL> likelihood = nll_real->getRooAbsL();
+      initMultiProcess(likelihood);
```

Comment:
```suggestion
      initMultiProcess(nll_real->getRooAbsL());
```
If `initMultiProcess` is taking a `const` shared pointer, you will not need this extra step :)